src/ui/Base/Component.ts

```diff
@@ -95,12 +95,10 @@ export class ComponentEvents {
   public on(el: HTMLElement | Window | Document, event: string, handler: Function);
   public on(el: Dom, event: string, handler: Function);
   public on(arg: any, event: string, handler: Function) {
-    if (!JQueryUtils.getJQuery() || !JQueryUtils.isInstanceOfJQuery(arg)) {
-      var htmlEl: HTMLElement = arg;
-      $$(htmlEl).on(event, this.wrapToCallIfEnabled(handler));
+    if (this.shouldTreatElementAsJQuery(arg)) {
```

Comment:
Since the two branches of the `if`/`else` only differ in the `arg` type, we could reduce the repetition by transforming it in a separate step. e.g.

```
const el = this.shouldTreatElementAsJQuery(arg) ? arg : $$(arg);
el.on(event, this.wrapToCallIfEnabled(handler))
```

Same comment for the other blocks.